### ISSUE TITLE
Static init.zsh script does not support `if then else fi` logic 

### ISSUE THREAD

**Issue by ericbn** [zimfw/zimfw#355]
[As reported](https://github.com/zimfw/zimfw/pull/323#issuecomment-571285700) by @AtomicCoding:

I've switched to the develop branch about a week ago, and everything so far has been amazing, it's really great to see how far Zim has come. However, I've been having a mild issue with how Zim precompiles things into one script. I have this line in my .zimrc:
```
if (( ${+TERM_PROGRAM} )); then
  if [ $TERM_PROGRAM = "iTerm.app" ]; then
    zmodule romkatv/powerlevel10k
  else
    zmodule minimal
  fi
else
  zmodule minimal
fi
```
And the expected behavior would be that when I run iTerm (which has a nerd font installed), I'd get a powerlevel10k prompt but when SSHing or using any other terminal, I'd get minimal. However, due to the way it seems to be compiled (I'm assuming it just checks which modules have been defined, and loads them), I've had to run `zimfw update` each time I switch from different terminals to make sure the prompt changes. This isn't a significant issue, and would likely require some work to copy over to the pre-compiled script, but it's just something I've been experiencing.

_Originally posted by @AtomicCoding in https://github.com/zimfw/zimfw/pull/323#issuecomment-571285700_

**Comment by ericbn** [comment]
[As commented](https://github.com/zimfw/zimfw/pull/323#issuecomment-571295544) by @PatTheMav:

@AtomicCoding Yeah if I'm not mistaken that kind of dynamic loading isn't really possible with new `zim` anymore as the module configuration is transpiled into an `init.zsh` containing all necessary `fpath`s, `autoloaders and `source` calls with the corresponding modules fetched from Github and pre-compiled.

As far as I can see there is no way to "select" both modules, but _not_ automatically sourcing them (because that's what `zim` will do, as the prompts will have `.zsh-theme` files) so you could have both managed by `zim`, but then `source` them yourself depending on `$TERM` e.g. in your `.zshrc`.

@ericbn any idea here? The most-straightforward option would be an option for `zmodule` that excludes them from autoloading (e.g. `--source /dev/null`). Bit of an edge-case though..

The only other way would be for a custom module that handles the switch based on `$TERM` and downloads/manages/sources the themes inside its own scope.

_Originally posted by @PatTheMav in https://github.com/zimfw/zimfw/pull/323#issuecomment-571295544_

**Comment by ericbn** [comment]
@AtomicCoding, moved the discussion to a new issue. Also, as a side comment, I see your're using @PatTheMav's minimal prompt that is hosted in the zimfw organization. @PatTheMav, just thinking about this again, it might not be really practical to have two separate repos for this code you're maintaining, Let me know your thoughts.

As for the reported issue, yeah, the `.zimrc` file is only executed during `zimfw` install/update/build/etc., so any logic you put there will be only valid to generate the static init.zsh script, not during the init.zsh script execution itself.

A solution I can think for this is to add a `--guard` (or maybe a better name, this was the first one that came to my mind) parameter to `zmodule`, so you could declare:

```
zmodule romkatv/powerlevel10k --guard '[[ ${TERM_PROGRAM} == iTerm.app ]]'
zmodule minimal --guard '[[ ${TERM_PROGRAM} != iTerm.app ]]'
```

It should be flexible enough and solve the issue.

**Comment by PatTheMav** [comment]
@ericbn Yeah haven't had time (or more specifically "made" time) to propose upstream changes to the original, but I'm fine with having an "official" version that just works and my own playground version.

Personally I've also switched to P10k, so any further changes might be limited, but I should make a spring cleaning effort. Good point.

**Comment by AtomicCoding** [comment]
@ericbn That sounds like a really good solution. I don't think this will be the last problem with custom logic in the static script, but that should definitely help fix a lot of them.

**Comment by Eriner** [comment]
Great idea, and I agree with using `--if` (or `--when`) rather than `--guard`, as it wasn't clear to me what `--guard` meant when skimming this issue and not reading the OP conditional.

**Comment by ericbn** [comment]
Released version 1.11.0 with this feature.
